include/openthread.h

```diff
@@ -191,6 +191,19 @@ otInstance *otInstanceInit(void);
 #endif
 
 /**
+ * This function initializes OpenThread stack.
+ *
+ * This function initializes OpenThread stack with following steps:
+ *   1. Initialize settings subsystem;
+ *   2. Retrieve active operational dataset from non-volatile memory, and apply the network configuration;
+ *   3. Retrieve pending operational dataset from non-volatile memory.
+ *
+ * @param[in] aInstance A pointer to an OpenThread instance.
+ *
+ */
+void otInit(otInstance *aInstance);
```

Comment:
I would actually put the logic of restoring the datasets from non-volatile memory in the `MeshCoP::DatasetManager` class or the `MeshCoP::ActiveDataset` and `MeshCoP::PendingDataset` classes.  The respective constructors can include the logic for restoring from non-volatile memory.  Then remove `otInit()` from the public API.